### PR TITLE
:mortar_board: Topic memory --- Introduce info/data and bits/bytes

### PR DESCRIPTION
**Pull request by jameshughes89**
### What

Clarify data vs info and bits/bytes. 

### Why

I should be clear on it somewhere, and this seems like the right spot for it before getting into storing the data. 

### Testing

:+1:

### CODE REVIEW

File: site/topics/memory/memory.rst [hunk 1]
```diff
@@ -2,16 +2,86 @@
...
+* Before getting into the details of memory, it's important to be clear on what data and information is
+
+    * Data is some collection of raw values/symbols that have no intrinsic meaning
+
+        * For example, the number :math:`3` is data
+
+
+    * Information is data that has meaning within some context
+
+        * :math:`3` is information on a quantity in the context of the question of how many dogs are in a car
```
Comment (twentylemon) [inline site/topics/memory/memory.rst]: I'd try to use less common words here. Outside of these definitions, data and information are used interchangably (erroneously) and it's bound to be a little confusing. Either that or really hammer home the difference, like [_easy_ vs _simple_](https://github.com/matthiasn/talk-transcripts/blob/master/Hickey_Rich/SimpleMadeEasy.md?plain=1#L23-L59) which took Rich 3 slides and nearly 10 minutes.

The students have some coding experience, so an example with some code may help, eg

```scala
// |--------| meaning (and partial context)
   val urMom = false
            // |----| data
// |----------------| information
```

Comment (jameshughes89) [reply site/topics/memory/memory.rst]: I'll think on this one on how best to approach it.

> data and information are used interchangably (erroneously) and it's bound to be a little confusing

Hell, I was working on the next section where I say that RAM in our typical computers hold data and instructions even though instructions are data.

Comment (jameshughes89) [reply site/topics/memory/memory.rst]: At the end of the day, the point of data vs information really is the difference between encoding vs meaning.

**Review by tmychaskiw-equinix — APPROVED** [review]

File: site/topics/memory/memory.rst [hunk 2]
```diff
@@ -2,16 +2,94 @@
...
+    * It could even be a pattern to control circuitry within a CPU to perform addition
+
+
+.. note::
+
+    The word "data" is used to refer to raw values/symbols that have no intrinsic meaning, while the word "information"
+    is used to refer to data that has some meaning within some context.
+
+    For example, the number :math:`3` alone is data, but the number :math:`3` can be information if it's referring to,
+    for instance, the quantity of ducks swimming in a pond.
```
Comment (twentylemon) [inline site/topics/memory/memory.rst:35]: I'd be careful here -- this is also data in the right context. In the context of computer memory, encodings are informational, but at higher levels it do be being data.

**Review by twentylemon — APPROVED** [review]

**Pull request merged**